services/cambio/getCambio.js

```diff
@@ -0,0 +1,18 @@
+import InternalError from '@/errors/InternalError';
+
+const axios = require('axios');
+
+export const getCambio = async (data, moeda) => {
+  try {
+    const resp = await axios.get(
+      `https://olinda.bcb.gov.br/olinda/servico/PTAX/versao/v1/odata/CotacaoMoedaDia(moeda=@moeda,dataCotacao=@dataCotacao)?@moeda='${moeda}'&@dataCotacao='${data}'&$top=100&$format=json`
+    );
+    return { info: resp.data };
+  } catch (error) {
+    return InternalError({
+      message: 'Não foi possível realizar a requisição',
+      type: 'internal_error',
+      name: 'INTERNAL_ERROR',
+    });
+  }
```

Comment:
Aqui você não precisa tratar a exceção, pois a `pages/api/cambio/v1/[moeda]/[data].js` já trata quando não é um erro esperado.

Mas se preferir realizar o tratamento mesmo assim sugiro fazer o seguinte:
```suggestion
  try {
    const resp = await axios.get(
      `https://olinda.bcb.gov.br/olinda/servico/PTAX/versao/v1/odata/CotacaoMoedaDia(moeda=@moeda,dataCotacao=@dataCotacao)?@moeda='${moeda}'&@dataCotacao='${data}'&$top=100&$format=json`
    );
    return { info: resp.data };
  } catch (error) {
    throw new InternalError({
      message: 'Não foi possível realizar a requisição',
      type: 'internal_error',
      name: 'INTERNAL_ERROR',
    });
  }
```